src/librustc_resolve/diagnostics.rs

```diff
@@ -197,7 +197,142 @@ See the Types section of the reference for more information about the primitive
 types:
 
 http://doc.rust-lang.org/reference.html#types
-"##
+"##,
+
+E0403: r##"
+Some type parameters have the same name. Example of erroneous code:
+
+```
+fn foo<T, T>(s: T, u: T) {} // error: the name `T` is already used for a type
+                            //        parameter in this type parameter list
+```
+
+Please verify that none of the type params are misspelled, and rename any
```

Comment:
params -> parameters

(I said params as a shorthand in the GH comment, not something that should have been copied verbatim :smile: )